pkg/controller/kubelet-config/helpers.go

```diff
@@ -184,14 +184,18 @@ func getManagedKubeletConfigKey(pool *mcfgv1.MachineConfigPool, client mcfgclien
 		if val != "" {
 			return fmt.Sprintf("99-%s-generated-kubelet-%s", pool.Name, val), nil
 		}
+		// the first kubeletconfig is being synced, return its default MC name
+		if ok && val == "" {
+			return ctrlcommon.GetManagedKey(pool, client, "99", "kubelet", getManagedKubeletConfigKeyDeprecated(pool))
+		}
```

Comment:
Might just be my personal preference, but I think it would have been easier for me to digest if this if block was combined with the first if block that returns the same thing and the relationship between the comments could then be clarified

Reply:
I changed the logic and removed the first block from this loop.